manim/mobject/types/vectorized_mobject.py

```diff
@@ -983,13 +983,13 @@ def point_from_proportion(self, alpha: float) -> np.ndarray:
             Point evaluated.
         """
 
-        curves_with_lengths = list(self.get_curve_functions_with_lengths())
+        if alpha >= 1:
+            return self.points[-1]
 
-        total_length = np.sum(length for _, length in curves_with_lengths)
-        target_length = alpha * total_length
+        target_length = alpha * self.get_arc_length()
         current_length = 0
 
-        for curve, length in curves_with_lengths:
+        for curve, length in self.get_curve_functions_with_lengths():
```

Comment:
Since `get_arc_length` uses `get_curve_functions_with_lengths` now, this will result in the lengths of the functions getting calculated twice in `point_from_proportion`, which is why I stored the curves and lengths in a list, since the lengths are needed to calculate the total length and to see when the accumulated length has reached the target length.

Reply:
Good point. I copied a slightly adjusted version of the `get_arc_length` code to `point_from_proportion` to allow reusing the lengths.